group/assignment6/RearrangeCars.cpp

```diff
@@ -0,0 +1,227 @@
+#include "RearrangeCars.h"
+
+RearrangeCars::RearrangeCars(const vector<int> &startStates) : startStates(startStates) {}
+
+void RearrangeCars::setStartStates(const vector<int> &startStates) {
+	this->startStates = startStates;
+}
+
+int RearrangeCars::findEmptySlotId() {
+	for (int slotId = 0; slotId < this->startStates.size(); slotId++) {
+		if (this->startStates[slotId] == EMPTY_SLOT_ID)
+			return slotId;  // No car has EMPTY_SLOT_ID id, it is reserved for empty slot.
+	}
+
+	return -1;  // This code should not be reached.
+}
+
+vector<int> RearrangeCars::findCarPositions(const vector<int> &currentStates) {
+	vector<int> carPositions(currentStates.size() - 1);
+	for (int i = 0; i < currentStates.size(); i++) {
+		if (currentStates[i] != EMPTY_SLOT_ID)  // Check for empty slot.
+			carPositions[currentStates[i]] = i;
+	}
+
+	return carPositions;
+}
+
+void RearrangeCars::moveCar(vector<int> &currentStates, vector<int> &carPosition, int from, int to) {
+	currentStates[to] = currentStates[from];
+	currentStates[from] = EMPTY_SLOT_ID;
+	carPosition[currentStates[to]] = to;
+}
+
+bool RearrangeCars::validMove(int carId, int emptySlotId, const vector<set<int>> *constraints) {
+	if (!constraints)  // Constraints are not set.

```

Comment:
Sometimes we put brackets around one-string if-s and for-s, sometimes we don't. I think one style should be chosen.